account-kit/core/src/actions/getSmartAccountClient.ts

```diff
@@ -218,12 +220,23 @@ export function getSmartAccountClient(
           isLoadingClient: false,
         };
       case "ModularAccountV2":
+        const mode = (accountParams as AccountConfig<"ModularAccountV2">).mode;
```

Comment:
If we avoid destructuring `accountParams` earlier, can we avoid the `as` casting here? It should propagate the type inference as being of `"ModularAccountV2:` thought the generic args.

Reply:
been fighting this for a bit but cannot seem to make it infer properly

Reply:
pushed a change to make it a bit safer at least